test/e2e/source_sinkbinding_v1alpha1_test.go

```diff
@@ -114,15 +121,29 @@ func TestSinkBindingDeployment(t *testing.T) {
 	// wait for all test resources to be ready
 	client.WaitForAllTestResourcesReadyOrFail()
 
-	// verify the logger service receives the event
+	// Look for events with expected data, and sinkbinding extension
 	expectedCount := 2
-	// Look for body.
-	if err := client.CheckLog(loggerPodName, lib.CheckerContainsAtLeast(data, expectedCount)); err != nil {
-		t.Fatalf("String %q does not appear at least %d times in logs of logger pod %q: %v", data, expectedCount, loggerPodName, err)
+	expectedSource := fmt.Sprintf("https://knative.dev/eventing/test/heartbeats/#%s/%s", client.Namespace, deploymentName)
+	matchFunc := func(ev ce.Event) bool {
```

Comment:
It would be cool to compose this match func in a similar way to the assertions I've designed here https://github.com/cloudevents/sdk-go/pull/522/files#diff-e32a0e715282f5a8765701ed1303641aR16 (not in this pr, just saying :smile: )